### PR TITLE
Support perf dashboard to run on php8 and PostgreSQL16.

### PR DESCRIPTION
**Pull request by dewei-zhu**
#### d114a6908521cd02077a4ef25e2c7cd8932f7732
<pre>
Support perf dashboard to run on php8 and PostgreSQL16.
<a href="https://bugs.webkit.org/show_bug.cgi?id=269050">https://bugs.webkit.org/show_bug.cgi?id=269050</a>
<a href="https://rdar.apple.com/100645242">rdar://100645242</a>

Reviewed by Alexey Proskuryakov and Ryosuke Niwa.

Adopt PostgreSQL12+ &apos;epoch from ...&apos; precision change and make it backward comaptible.
Grant &apos;CREATE&apos; permission to database user which is no longer granted by default since PostgreSQL 15.
Fix a build request url / status description cannot be updated after build request is completed which
was causing test case to fail.
Fix an invalid metric name in mock data.

* Websites/perf.webkit.org/public/api/build-requests.php: Fix a bug that
a build request url / status description cannot be updated after build request is completed.
* Websites/perf.webkit.org/public/api/measurement-set.php: PHP syntax update and adopt &apos;epoch from ...&apos;
precision change.
* Websites/perf.webkit.org/public/include/admin-header.php:
* Websites/perf.webkit.org/public/include/build-requests-fetcher.php:
* Websites/perf.webkit.org/public/include/commit-log-fetcher.php:
* Websites/perf.webkit.org/public/include/db.php:
* Websites/perf.webkit.org/public/include/manifest-generator.php:
* Websites/perf.webkit.org/public/include/report-processor.php:
* Websites/perf.webkit.org/public/include/repository-group-finder.php:
* Websites/perf.webkit.org/public/include/test-name-resolver.php:
* Websites/perf.webkit.org/public/include/test-path-resolver.php:
* Websites/perf.webkit.org/public/include/uploaded-file-helpers.php:
* Websites/perf.webkit.org/server-tests/resources/mock-data.js: Rename &apos;some metric&apos; to a valid metric
name &apos;Time&apos; which is checked on metric.js.
(MockData.addMockConfiguration):
(MockData.set addMockDataWithTestParameterSets):
(MockData.set addMockDataWithBuildAndTestTypeTestParameterSets):
* Websites/perf.webkit.org/server-tests/resources/test-server.js: Add query to grant &apos;CREATE&apos; permission to database
user.
(TestServer.prototype._ensureTestDatabase):
(TestServer.prototype._startApache):
* Websites/perf.webkit.org/tools/js/remote.js: Send &apos;Content-Length&apos; header.
(NodeRemoteAPI.prototype.sendHttpRequest):
* Websites/perf.webkit.org/unit-tests/measurement-set-analyzer-tests.js: Update test expectation accordingly.
* Websites/perf.webkit.org/unit-tests/resources/mock-v3-models.js: ename &apos;some metric&apos; to a valid metric
name &apos;Time&apos;.

Canonical link: <a href="https://commits.webkit.org/274413@main">https://commits.webkit.org/274413@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/ef1bb87e29bc4086ef08f6e0005521b31a6e9c2b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [⏳ 🧪 style ](https://ews-build.webkit.org/#/builders/Style-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 ios ](https://ews-build.webkit.org/#/builders/iOS-17-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 mac ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wincairo ](https://ews-build.webkit.org/#/builders/WinCairo-EWS "Waiting in queue, processing has not started yet") 
| [⏳ 🧪 bindings ](https://ews-build.webkit.org/#/builders/Bindings-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 ios-sim ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 mac-AS-debug ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Sonoma-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 wpe-wk2 ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 webkitperl ](https://ews-build.webkit.org/#/builders/WebKitPerl-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 ios-wk2 ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-mac ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 webkitpy ](https://ews-build.webkit.org/#/builders/WebKitPy-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 ios-wk2-wpt ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk1 ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 gtk ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🛠 🧪 jsc ](https://ews-build.webkit.org/#/builders/JSC-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-ios ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk2 ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 gtk-wk2 ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🛠 🧪 jsc-arm64 ](https://ews-build.webkit.org/#/builders/JSC-Tests-arm64-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 tv ](https://ews-build.webkit.org/#/builders/tvOS-17-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-AS-debug-wk2 ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Sonoma-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-gtk ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 services ](https://ews-build.webkit.org/#/builders/Services-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 tv-sim ](https://ews-build.webkit.org/#/builders/tvOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk2-stress ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 jsc-armv7 ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [⏳ 🛠 watch ](https://ews-build.webkit.org/#/builders/watchOS-10-Build-EWS "Waiting in queue, processing has not started yet") | | [⏳ 🧪 jsc-armv7-tests ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5097 "Built successfully and passed tests") | [⏳ 🛠 watch-sim ](https://ews-build.webkit.org/#/builders/watchOS-10-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | |

### CODE REVIEW

File: Websites/perf.webkit.org/public/include/uploaded-file-helpers.php [hunk 1]
```diff
@@ -194,4 +194,4 @@ function prune_old_files($db, $size_needed, $remote_user)
     return FALSE;
 }
 
-?>
+?>
```
Comment (rniwa) [inline Websites/perf.webkit.org/public/include/uploaded-file-helpers.php:197]: Are you sure this line break addition is intended?

**Review by rniwa — APPROVED** [review]

File: Websites/perf.webkit.org/public/include/uploaded-file-helpers.php (hunk 1 above)
Comment (dewei-zhu) [reply Websites/perf.webkit.org/public/include/uploaded-file-helpers.php:197]: This is actually removes `\n`
```
@@ -194,4 +194,4 @@ function prune_old_files($db, $size_needed, $remote_user)
     return FALSE;
 }
 
-?>
+?>
\ No newline at end of file
```

Comment (dewei-zhu) [reply Websites/perf.webkit.org/public/include/uploaded-file-helpers.php:197]: Will have a separate change to use open php tags anyway.

File: Websites/perf.webkit.org/public/api/build-requests.php [hunk 2]
```diff
@@ -62,8 +62,17 @@ function update_builds($db, $updates) {
                 $db->rollback_transaction();
                 exit_with_error('FailedToFindBuildRequest', array('buildRequest' => $id));
             }
-            if ($request_row['request_status'] == 'completed')
+            if ($request_row['request_status'] == 'completed') {
+                $filtered_fields_to_update = array();
+                foreach($fields_to_update as $name => $value) {
```
Comment (aproskuryakov) [inline Websites/perf.webkit.org/public/api/build-requests.php]: I think that there should be a space after `foreach`.

File: Websites/perf.webkit.org/public/api/measurement-set.php [hunk 3]
```diff
@@ -187,7 +196,10 @@ static function format_map()
     }
 
     private static function format_run(&$run, &$commit_time) {
-        $commit_time = intval($run['revision_time']);
+        # Value precision of 'epoch from ...' can vary between different versions of postgres.
+        # 1538041792670.999 will be 1538041792671 on Postgres 11 or earlier, and 1538041792670.999 on later versions.
+        # Round it up to keep compatibility.
```
Comment (aproskuryakov) [inline Websites/perf.webkit.org/public/api/measurement-set.php]: This comment would be even better if it explained why need to keep compatibility. Presumably this has to do either with other code breaking if we don't round up, or with existing content in database - but what exactly breaks if we don't?

Also, I'm a bit confused by the "up" part. Surely `.999` will be rounded up, but `.001` will be rounded down, if we get that?

**Review by aproskuryakov — APPROVED** [review]

File: Websites/perf.webkit.org/public/api/measurement-set.php (hunk 3 above)
Comment (dewei-zhu) [reply Websites/perf.webkit.org/public/api/measurement-set.php]: OK, this is intended to fix unit test added by https://commits.webkit.org/206368@main still works on newer version of Postgres. Will rephrase and provide another round down example.

**Pull request merged**